lib/lte_link_control/Kconfig

```diff
@@ -23,6 +23,26 @@ config HEAP_MEM_POOL_ADD_SIZE_LTE_LINK_CONTROL
 
 # Modules: these Kconfig options enable specific features of the library.
 
+config LTE_LC_NETWORK_REGISTRATION_MODULE
+	bool "Network registration module"
+	default y
+
+config LTE_LC_FUNCTIONAL_MODE_MODULE
```

Comment:
There may be dependencies to these core modules. At least `cfun.c` currently subscribes to notifications from the modem when transitioning to functional modes 1, 2 and 21. We could get rid of that by moving those to `lte_lc_modem_hooks.c`, I think, but coneval.c and cereg.c are also using `cfun_mode_get()` and c`fun_modem_set()`, so I don't think you can disable `LTE_LC_FUNCTIONAL_MODE_MODULE`.